### PR TITLE
doc: add missing documentation in PR76408

### PR DESCRIPTION
**Pull request by valeriosetti**
PR #76408 introduced some kconfig changes (some symbols were renamed while a new one was added), but none of these were documented. This PR covers this gap amending both migration guide and release notes.

### CODE REVIEW

**Review by dkalowsk — DISMISSED** [review]

File: doc/releases/migration-guide-4.0.rst [hunk 1]
```diff
@@ -40,6 +40,9 @@ Mbed TLS
 
 * The Kconfig options ``CONFIG_MBEDTLS_TLS_VERSION_1_0`` and ``CONFIG_MBEDTLS_TLS_VERSION_1_1``
   have been removed because Mbed TLS doesn't support TLS 1.0 and 1.1 anymore since v3.0. (:github:`76833`)
+* Following Kconfig symbols were renamed (:github:`76408`):
+  * ``CONFIG_MBEDTLS_ENTROPY_ENABLED`` is now ``CONFIG_MBEDTLS_ENTROPY_C``,
```
Comment (tomi-font) [inline doc/releases/migration-guide-4.0.rst:44]:
```suggestion
* The following Kconfig symbols were renamed (:github:`76408`):
  * ``CONFIG_MBEDTLS_ENTROPY_ENABLED`` is now :kconfig:option:`CONFIG_MBEDTLS_ENTROPY_C`,
```

File: doc/releases/release-notes-4.0.rst [hunk 2]
```diff
@@ -315,6 +315,11 @@ Libraries / Subsystems
 
   * Mbed TLS was updated to version 3.6.1. The release notes can be found at:
     https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.1
+  * New Kconfig symbol :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`
+    was added to allow ``psa_get_random()`` to make use of non cryptographically
```
Comment (tomi-font) [inline doc/releases/release-notes-4.0.rst:319]:
```suggestion
    was added to allow ``psa_get_random()`` to make use of non-cryptographically
```

File: doc/releases/release-notes-4.0.rst [hunk 3]
```diff
@@ -315,6 +315,11 @@ Libraries / Subsystems
...
+    secure random sources when :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG`
+    is also set. This is only meant to be used for test purposes, not in production.
```
Comment (tomi-font) [inline doc/releases/release-notes-4.0.rst:321]:
```suggestion
    is also enabled. This is only meant to be used for test purposes, not in production.
```

File: doc/releases/release-notes-4.0.rst [hunk 4]
```diff
@@ -315,6 +315,11 @@ Libraries / Subsystems
 
   * Mbed TLS was updated to version 3.6.1. The release notes can be found at:
     https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.1
+  * New Kconfig symbol :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`
```
Comment (tomi-font) [inline doc/releases/release-notes-4.0.rst:318]:
```suggestion
  * A new Kconfig symbol (:kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`)
```
or
```suggestion
  * The Kconfig symbol :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`
```

**Review by tomi-font — CHANGES_REQUESTED** [review]
Great!

**Review by tomi-font — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]
Can the title of the commit be changed to mention about the new Kconfig added instead of stating it is adding missing documentation for a PR.

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]
Request to update the commit title

**Comment by valeriosetti** [comment]
> Can the title of the commit be changed to mention about the new Kconfig added instead of stating it is adding missing documentation for a PR.

Sure, I just did this. I split the single commit into 2 because it was easier to add the proper commits' title then.

**Review by mmahadevan108 — APPROVED** [review]

File: doc/releases/migration-guide-4.0.rst [hunk 5]
```diff
@@ -40,6 +40,9 @@ Mbed TLS
 
 * The Kconfig options ``CONFIG_MBEDTLS_TLS_VERSION_1_0`` and ``CONFIG_MBEDTLS_TLS_VERSION_1_1``
   have been removed because Mbed TLS doesn't support TLS 1.0 and 1.1 anymore since v3.0. (:github:`76833`)
+* The following Kconfig symbols were renamed (:github:`76408`):
+  * ``CONFIG_MBEDTLS_ENTROPY_ENABLED`` is now :kconfig:option:``CONFIG_MBEDTLS_ENTROPY_C``,
+  * ``CONFIG_MBEDTLS_ZEPHYR_ENTROPY`` is now :kconfig:option:``CONFIG_MBEDTLS_ENTROPY_POLL_ZEPHYR``.
```
Comment (kartben) [inline doc/releases/migration-guide-4.0.rst:45]: I'll fix this as part of 78266 but note that the kconfig:option: role calls for single backticks, not double. Luckily, these kind of easy-to-make mistakes will soon be caught by sphinx-lint!

**Pull request merged**